File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string: str):
        self.rows = [[int(x) for x in row.split(' ')] for row in matrix_string.split('\n')]
        self.columns = list(map(list, zip(*self.rows)))

    def row(self, index: int) -> list:
        return self.rows[index - 1]

    def column(self, index: int) -> list:
       	return self.columns[index - 1]

File: java/matrix/src/main/java/Matrix.java
import java.util.Arrays;
import java.util.stream.IntStream;

class Matrix
{
    private int[][] rows;
    private int[][] columns;

    Matrix(String matrixAsString)
    {
        this.rows = Arrays.stream(matrixAsString.split("\\n"))
                .map(this::toIntArray)
                .toArray(int[][]::new);

        this.columns = IntStream.range(0, this.rows[0].length)
                .mapToObj(this::createColumn)
                .toArray(int[][]::new);
    }

    private int[] toIntArray(String numbers)
    {
        return Arrays.stream(numbers.split("\\s"))
                .mapToInt(Integer::valueOf)
                .toArray();
    }

    private int[] createColumn(int columnNumber)
    {
        return Arrays.stream(this.rows)
                .mapToInt(row -> row[columnNumber])
                .toArray();
    }

    int[] getRow(int rowNumber)
    {
        return this.rows[rowNumber - 1];
    }

    int[] getColumn(int columnNumber)
    {
        return this.columns[columnNumber - 1];
    }
}
